simulator/host_system.go

```diff
@@ -61,8 +61,10 @@ func NewHostSystem(host mo.HostSystem) *HostSystem {
 	hardware := *host.Summary.Hardware
 	hs.Summary.Hardware = &hardware
 
-	info := *esx.HostHardwareInfo
-	hs.Hardware = &info
+	if hs.Hardware == nil {
+		info := *esx.HostHardwareInfo
+		hs.Hardware = &info
+	}
```

Comment:
question: could this be simplified to:

```suggestion
	if hs.Hardware == nil {
		hs.Hardware = *esx.HostHardwareInfo
	}
```

Reply:
Ah, sorry seems like this is supposed to be a copy of the object. Any DeepCopy concerns on that struct (haven't checked the struct fields)?

Reply:
Shallow copy is enough, the only field we ever change is the Uuid: https://github.com/vmware/govmomi/blob/afc67f589a3d1bcb066ac2a0724a1cace1b3f4a9/simulator/host_system.go#L103

I'll add a comment on that w/ the next push.